File: sdks/java/src/main/java/com/boltprediction/sdk/model/LabelImportDbRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 数据库标注导入请求 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class LabelImportDbRequest {

    @JsonProperty("source_table")
    private String sourceTable;

    @JsonProperty("node_type")
    private String nodeType;

    @JsonProperty("id_field")
    private String idField;

    @JsonProperty("label_field")
    private String labelField;

    @JsonProperty("data_field")
    private Object dataField;

    @JsonProperty("timestamp_field")
    private Object timestampField;

    @JsonProperty("where_clause")
    private Object whereClause;

    @JsonProperty("labeler_name")
    private Object labelerName;

    @JsonProperty("auto_approve")
    private Boolean autoApprove;

    public String getSourceTable() {
        return sourceTable;
    }

    public void setSourceTable(String sourceTable) {
        this.sourceTable = sourceTable;
    }

    public String getNodeType() {
        return nodeType;
    }

    public void setNodeType(String nodeType) {
        this.nodeType = nodeType;
    }

    public String getIdField() {
        return idField;
    }

    public void setIdField(String idField) {
        this.idField = idField;
    }

    public String getLabelField() {
        return labelField;
    }

    public void setLabelField(String labelField) {
        this.labelField = labelField;
    }

    public Object getDataField() {
        return dataField;
    }

    public void setDataField(Object dataField) {
        this.dataField = dataField;
    }

    public Object getTimestampField() {
        return timestampField;
    }

    public void setTimestampField(Object timestampField) {
        this.timestampField = timestampField;
    }

    public Object getWhereClause() {
        return whereClause;
    }

    public void setWhereClause(Object whereClause) {
        this.whereClause = whereClause;
    }

    public Object getLabelerName() {
        return labelerName;
    }

    public void setLabelerName(Object labelerName) {
        this.labelerName = labelerName;
    }

    public Boolean getAutoApprove() {
        return autoApprove;
    }

    public void setAutoApprove(Boolean autoApprove) {
        this.autoApprove = autoApprove;
    }

}

File: sdks/python/bolt_prediction_sdk/models/label_import_db_request.py
"""
LabelImportDBRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class LabelImportDbRequest(SDKBaseModel):
    """数据库标注导入请求"""

    source_table: str = Field(description="源表名")
    node_type: str = Field(description="节点类型: bolt/flange")
    id_field: str = Field(description="节点ID字段名")
    label_field: str = Field(description="标签字段名")
    data_field: Optional[Any] = Field(description="数据点字段名", default=None)
    timestamp_field: Optional[Any] = Field(description="时间戳字段名", default=None)
    where_clause: Optional[Any] = Field(description="WHERE条件，不带WHERE关键字", default=None)
    labeler_name: Optional[Any] = Field(description="标注人姓名", default=None)
    auto_approve: Optional[bool] = Field(description="是否自动审核通过", default=True)
